### DISCUSSION TITLE
Too long time to finish the task

### DISCUSSION THREAD

**Discussion by aquarelacs-arthur-rosa** [GitTools/GitVersion#3634] [Q&A]
**Describe the bug**
I have a repository that uses Git Version for two years, currently the task time in Azure DevOps takes 10 minutes, is there a possibility to reduce this time?

## Expected Behavior
Run in a few seconds

## Actual Behavior
Currently it takes 10 to 15 minutes just for the task.

![image](https://github.com/GitTools/GitVersion/assets/133425456/db016b55-a1c9-42fc-9806-40aa3705d5d9)

## Possible Fix
Any configuration in Yaml to solve the problem?

## Steps to Reproduce
To reproduce the bug is a little complicated because it is necessary to have a repository of more than 2 years and have a huge amount of PRs and commits

## Context
This bug affects every time Ci's pipeline runs, extending its time by 100 times.

## Your Environment
My environment used is Azure DevOps, below is the GitVersion Yaml

## GitVersion
steps:
- task: gittools.gitversion.gitversion-task.GitVersion@5
  displayName: GitVersion
  inputs:
    runtime: full

## Full Ci
pool:
  name: Azure Pipelines
#Your build pipeline references an undefined variable named ‘git diff HEAD HEAD~ --name-only -r’. Create or edit the build pipeline for this YAML file, define the variable on the Variables tab. See https://go.microsoft.com/fwlink/?linkid=865972

steps:
- task: gittools.gitversion.gitversion-task.GitVersion@5
  displayName: GitVersion
  inputs:
    runtime: full

- script: |
   7z a tools.zip tools
   7z a login.zip login
  displayName: 'Zip Folders'

- task: CopyFiles@2
  displayName: 'Copy Files to: $(build.artifactstagingdirectory)'
  inputs:
    Contents: |
     **
     !**\.git\**
     !**\componentes\**
     !**\login\**
     !**\tools\**
    TargetFolder: '$(build.artifactstagingdirectory)'

- task: PublishPipelineArtifact@1
  displayName: 'Publish Pipeline Artifact'
  inputs:
    targetPath: '$(Build.ArtifactStagingDirectory)'
    artifact: drop

![image](https://github.com/GitTools/GitVersion/assets/133425456/627848e8-83c3-44c5-8608-b0705d5aa413)

- Version Used: gittools.gitversion.gitversion-task.GitVersion@5
- Link to your project: is private.
- Link to your CI build (if appropriate): is private.

**Comment by HHobeck** [comment]
First I would like to thank you for reporting your experience. I have seen some issues (like https://github.com/GitTools/GitVersion/issues/3212) which are related to performance. It would be good to provide any information about your environment and how your branching and merging strategy is.

May I ask you to help us with the following one or more points:

- Give background information about your context/environment
- Find out where the problem might be located (Is it because of so many tags? Is it because of so many branches? Is it because you are using a special mode?)
- Create a repository which can be used for analyzing and improving the performance
- Statistic how the complexity is in different circumstances (https://en.wikipedia.org/wiki/Time_complexity)

BTW: There is already a pre-release (see version 6.x or main branch) existing which includes a lot of changes. At this moment it would make no sense to invest time on version 5.0.1.

**Reply by aquarelacs-arthur-rosa** [reply]
In azure devops for tasks only version 5.0.X appears to select, about our merge strategy, we use the following way.

develop, homologation, master.

We develop in develop with a master base, as soon as we finalize the merge for approval, we use a lot of PRs and merge.

Does this already help? I'm sorry for the delay!.

**Comment by HHobeck** [comment]
The bug description is not explicit enough to break out improvements or other tasks. Thus I'm going to move this issue to the discussion section.

**Comment by aquarelacs-arthur-rosa** [comment]
@HHobeck I will be available if you have any further questions.

**Reply by HHobeck** [reply]
It makes no sense to me to invest time on version 5.0.x because this version will be replaced with the version 6.x in the near future. If you want to help the project you need to switch to the pre-release version and create a repository to reproduce this behavior or write integration tests.

BTW: You are free to engage people like me to support you.
